pkg/controller/clusterdeployment/clusterdeployment_controller.go

```diff
@@ -1144,6 +1165,10 @@ func (r *ReconcileClusterDeployment) ensureManagedDNSZoneDeleted(cd *hivev1.Clus
 }
 
 func (r *ReconcileClusterDeployment) syncDeletedClusterDeployment(cd *hivev1.ClusterDeployment, cdLog log.FieldLogger) (reconcile.Result, error) {
+	if controllerutils.IsDeleteProtected(cd) {
```

Comment:
Should we include r.protectedDelete in the if? Maybe you had it enabled but turned it off later?

Reply:
The only thing that enabling protected delete in the HiveConfig does is determine whether Hive will automatically add the annotation on the ClusterDeployment. The user can still add the annotation manually.